### PR TITLE
Add Repositories list in Function view

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:

As in title: Add Repositories list in Function view and tests for that.

**Related issue(s)**
https://github.com/kyma-project/console/issues/1932

### CODE REVIEW

**Review by aerfio — CHANGES_REQUESTED** [review]
There are few bugs:
- after changing anything in that modal "create" button becomes enabled, when there's no repository CR
image
- css:
image
- this one is buggy, it allows me to create fn, bug crashes on cbs side
image
- besides, to see those errors I had to type in something and then remove that...
image
- we'd have to add logs from that job, that fetches gitrepo to that minimalised LogsUI, because if I supply wrong values I have no way to tell what went wrong

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/RepositoryConfig/RepositoryConfig.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    }
+  }, [errors, setIsValid, setEditStatus, baseDirRef, referenceRef, lambda]);
+
+  function validateReference(reference, setStatus) {
+    if (!reference) {
+      setStatus(REPOSITORY_CONFIG_PANEL.INPUTS.REFERENCE.ERRORS.EMPTY);
+      addError('reference');
+      return;
+    }
+    setStatus('');
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/RepositoryConfig/RepositoryConfig.js:76]: I don't understand

File: core-ui/src/components/Lambdas/components/DropdownInput/DropdownInput.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  label = '',
+  inlineHelp = undefined,
+  defaultValue = '',
+  onChange = () => void 0,
+  options = [],
+}) => {
+  if (!options.length) {
+    return null;
+  }
+  const df = defaultValue === undefined ? options[0].value : defaultValue;
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/components/DropdownInput/DropdownInput.js:20]: > `and what if options[0] doesn't exist` 

See the `if` above

> and you've set default value of defaultValue (use more descriptive name) to "", so undefined supplied to that component is always overridden by ""

It's an intentional logic :) Sometimes form needs a empty string (not null or undefined).

File: core-ui/src/components/Lambdas/constants.js [hunk 3]
```diff
@@ -82,17 +86,127 @@ export const LAMBDAS_LIST = {
...
-        INLINE_HELP: `The key/value pair should be separated by '=', consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character. The key cannot be empty.`,
-      },
       RUNTIME: {
-        NAME: 'Runtime',
+        LABEL: 'Runtime',
+        INLINE_HELP: `Select a runtime on which the Function will run.`,
+      },
+      SOURCE_TYPE: {
+        LABEL: 'Source type',
+        INLINE_HELP: `Select type of Function's source. Available: Plain (code and dependencies is given as plain text) or Git (code and dependencies is fetched from git repository).`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:95]:
```suggestion
        INLINE_HELP: `Select the type of your Function's code source: Inline editor (code and dependencies are provided as plain text) or Git repository (code and dependencies are fetched from a Git repository).`,
```

File: core-ui/src/components/Lambdas/constants.js [hunk 4]
```diff
@@ -82,17 +86,127 @@ export const LAMBDAS_LIST = {
...
+        {
+          KEY: 'Public',
+          VALUE: '',
+        },
+        {
+          KEY: 'Basic',
+          VALUE: 'basic',
+        },
+        {
+          KEY: 'Key',
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:186]:
```suggestion
          KEY: 'SSH key',
```

File: core-ui/src/components/Lambdas/constants.js [hunk 5]
```diff
@@ -346,6 +486,38 @@ export const CODE_AND_DEPENDENCIES_PANEL = {
...
+  },
+  ERRORS: {
+    INVALID: 'At least one field is empty.',
+    NO_CHANGES: 'No changes made.',
+  },
+  INPUTS: {
+    REFERENCE: {
+      LABEL: 'Reference',
+      INLINE_HELP: `Reference to changes - branch or revision of commit.`,
+      PLACEHOLDER: 'Reference',
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:505]:
```suggestion
      PLACEHOLDER: 'Enter a branch name or commit revision',
```

File: core-ui/src/components/Lambdas/constants.js [hunk 6]
```diff
@@ -82,17 +86,127 @@ export const LAMBDAS_LIST = {
...
+        ],
+      },
+      REPOSITORY: {
+        LABEL: 'Repository',
+        INLINE_HELP: `Repository which you want use.`,
+      },
+      REFERENCE: {
+        LABEL: 'Reference',
+        INLINE_HELP: `Reference to changes - branch or revision of commit.`,
+        PLACEHOLDER: 'Enter a reference',
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:114]:
```suggestion
        PLACEHOLDER: 'Enter a branch name or commit revision',
```

File: core-ui/src/components/Lambdas/constants.js [hunk 7]
```diff
@@ -82,17 +86,127 @@ export const LAMBDAS_LIST = {
...
+        ERRORS: {
+          EMPTY: 'Reference is required.',
+        },
+      },
+      BASE_DIR: {
+        LABEL: 'Base directory',
+        INLINE_HELP: `Directory where are located the Function's code and dependencies.`,
+        PLACEHOLDER: 'Enter a base directory',
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:122]:
```suggestion
        PLACEHOLDER: 'Enter a base directory',
```

File: core-ui/src/components/Lambdas/constants.js [hunk 8]
```diff
@@ -179,6 +299,10 @@ export const GQL_MUTATIONS = {
       SUCCESS_MESSAGE: `Code and dependencies for Function "{lambdaName}" were successfully updated`,
       ERROR_MESSAGE: `Couldn't update code and dependencies for Function "{lambdaName}" due to this error: {error}`,
     },
+    REPOSITORY_CONFIG: {
+      SUCCESS_MESSAGE: `Repository configuration for Function "{lambdaName}" were successfully updated`,
```
[the remark this answers is not in the document]
Comment (kazydek) [reply core-ui/src/components/Lambdas/constants.js:303]: I would add commas at the end of all `SUCCESS_MESSAGE` & `ERROR_MESSAGE`.

File: core-ui/src/components/Lambdas/constants.js [hunk 9]
```diff
@@ -82,17 +86,127 @@ export const LAMBDAS_LIST = {
...
+        TOO_LONG: 'Repository name cannot be longer than 63 characters.',
+      },
+    },
+  },
+};
+
+export const REPOSITORY_AUTH = {
+  PUBLIC: 'Public',
+  BASIC: 'Basic',
+  KEY: 'Key',
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:207]:
```suggestion
  KEY: 'SSH key',
```

File: core-ui/src/components/Lambdas/constants.js [hunk 10]
```diff
@@ -82,17 +86,127 @@ export const LAMBDAS_LIST = {
...
+          'There is already a Repository with the same name in this Namespace.',
+        TOO_LONG: 'Repository name cannot be longer than 63 characters.',
+      },
+    },
+    URL: {
+      LABEL: 'URL',
+      INLINE_HELP: `The URL must be valid a git url and must start with one of the following prefixes: http(s), git, ssh`,
+      PLACEHOLDER: 'Enter an URL for a Repository',
+      ERRORS: {
+        INVALID: `Invalid Repository URL. The URL must start with one of the following prefixes: http(s), git, ssh.`,
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/constants.js:169]:
```
Invalid Repository URL. The URL must starts with the "http(s)", "git", or "ssh" prefix and ends with the ".git" suffix.
```

File: core-ui/src/components/Lambdas/constants.js [hunk 11]
```diff
@@ -196,6 +320,22 @@ export const GQL_MUTATIONS = {
       MESSAGE: `Are you sure you want to delete Function "{lambdaName}" and all related resources?`,
     },
   },
+  CREATE_REPOSITORY: {
+    SUCCESS_MESSAGE: `Repository "{repositoryName}" was successfully created`,
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/constants.js:324]: We had `successfully` in all success messages :)

File: core-ui/src/components/Lambdas/constants.js [hunk 12]
```diff
@@ -196,6 +320,22 @@ export const GQL_MUTATIONS = {
...
+    ERROR_MESSAGE: `Couldn't create Repository "{repositoryName}" due to this error: {error}`,
+  },
+  UPDATE_REPOSITORY: {
+    SUCCESS_MESSAGE: `Repository "{repositoryName}" was successfully updated`,
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/constants.js:328]: We had successfully in all success messages :)

File: core-ui/src/components/Lambdas/constants.js [hunk 13]
```diff
@@ -196,6 +320,22 @@ export const GQL_MUTATIONS = {
...
+    ERROR_MESSAGE: `Couldn't update Repository "{repositoryName}" due to this error: {error}`,
+  },
+  DELETE_REPOSITORY: {
+    SUCCESS_MESSAGE: `Repository "{repositoryName}" was successfully deleted`,
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/constants.js:332]: We had successfully in all success messages :)

File: core-ui/src/components/Lambdas/constants.js [hunk 14]
```diff
@@ -82,17 +86,127 @@ export const LAMBDAS_LIST = {
...
+          },
+        ],
+      },
+      REPOSITORY: {
+        LABEL: 'Repository',
+        INLINE_HELP: `Repository which you want use.`,
+      },
+      REFERENCE: {
+        LABEL: 'Reference',
+        INLINE_HELP: `Reference to changes - branch or revision of commit.`,
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/constants.js:113]: Reference for source in repository :) branch or commit revision.

File: core-ui/src/components/Lambdas/constants.js [hunk 15]
```diff
@@ -82,17 +86,132 @@ export const LAMBDAS_LIST = {
...
         },
       },
-      LABEL: {
-        LABEL: 'Labels',
-        INLINE_HELP: `The key/value pair should be separated by '=', consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character. The key cannot be empty.`,
-      },
       RUNTIME: {
-        NAME: 'Runtime',
+        LABEL: 'Runtime',
+        INLINE_HELP: `Select a Runtime on which your Function will run`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:91]:
```suggestion
        INLINE_HELP: `Runtime on which your Function will run`,
```

File: core-ui/src/components/Lambdas/constants.js [hunk 16]
```diff
@@ -82,17 +86,132 @@ export const LAMBDAS_LIST = {
...
+      },
+      SOURCE_TYPE: {
+        LABEL: 'Source type',
+        INLINE_HELP: `Select the type of your Function's code source: Inline editor (code and dependencies are provided as plain text) or Git repository (code and dependencies are fetched from a Git repository).`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:95]:
```suggestion
        INLINE_HELP: `Type of your Function's code source: Inline editor (code and dependencies are provided as plain text) or Git repository (code and dependencies are fetched from a Git repository)`,
```

File: core-ui/src/components/Lambdas/constants.js [hunk 17]
```diff
@@ -82,17 +86,132 @@ export const LAMBDAS_LIST = {
...
+      INLINE_HELP: `URL must be a valid Git repository address that starts with the "http(s)", "git", or "ssh" prefix and ends with the ".git" suffix.`,
+      PLACEHOLDER: 'Enter the URL address of your Git repository (Required)',
+      ERRORS: {
+        INVALID: `Invalid Repository URL. The URL must start with one of the following prefixes: http(s), git, ssh.`,
+        EMPTY: 'Repository URL is required.',
+      },
+    },
+    AUTH_TYPE: {
+      LABEL: 'Authorization',
+      INLINE_HELP: `Decide if you want to authenticate with your Git repository using token and password (Basic), authentication key (SSH key), or if you don't want to use any authorization method (Public).`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:180]:
```suggestion
      INLINE_HELP: `Authenticate with your Git repository using a token or password (Basic), an authentication key (SSH key), or don't use any authorization method (Public)`,
```

File: core-ui/src/components/Lambdas/constants.js [hunk 18]
```diff
@@ -167,56 +292,76 @@ export const GQL_QUERIES = {
...
+      MESSAGE: `Are you sure you want to delete Repository "{repositoryName}" and all related resources?`,
+    },
+  },
   CREATE_TRIGGERS: {
-    SUCCESS_MESSAGE_SINGLE: `Event Trigger created successfully`,
-    SUCCESS_MESSAGE_MANY: `Event Triggers created successfully`,
-    ERROR_MESSAGE_SINGLE: `Event Trigger for Function "{lambdaName}" couldn't be created due to this error: {error}`,
-    ERROR_MESSAGE_MANY: `Couldn't create Event Triggers for Function "{lambdaName}" due to this error: {error}`,
+    SUCCESS_MESSAGE_SINGLE: `Event Trigger created successfully.`,
+    SUCCESS_MESSAGE_MANY: `Event Triggers created successfully.`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:346]:
```suggestion
    SUCCESS_MESSAGE_MANY: `Event Triggers created.`,
```

Comment (kazydek) [reply core-ui/src/components/Lambdas/constants.js:346]: I would remove "successfully" everywhere - it's not necessary and not recommended in guides. I know we had it everywhere before but it's time for changes :)

File: core-ui/src/components/Lambdas/constants.js [hunk 19]
```diff
@@ -82,17 +86,132 @@ export const LAMBDAS_LIST = {
...
-        LABEL: 'Labels',
-        INLINE_HELP: `The key/value pair should be separated by '=', consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character. The key cannot be empty.`,
-      },
       RUNTIME: {
-        NAME: 'Runtime',
+        LABEL: 'Runtime',
+        INLINE_HELP: `Select a Runtime on which your Function will run`,
+      },
+      SOURCE_TYPE: {
+        LABEL: 'Source type',
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:94]:
```suggestion
        LABEL: 'Source Type',
```

File: core-ui/src/components/Lambdas/constants.js [hunk 20]
```diff
@@ -82,17 +86,132 @@ export const LAMBDAS_LIST = {
...
+    CONFIRM_BUTTON: {
+      TEXT: 'Update',
+    },
+  },
+  MODAL_ERROR: 'At least one field is empty or incorrectly formatted.',
+  MODAL_INPUTS: {
+    NAME: {
+      ERRORS: {
+        EMPTY: 'Repository name is required.',
+        INVALID: `Invalid Repository URL. The URL must starts with the "http(s)", "git", or "ssh" prefix and ends with the ".git" suffix.`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:163]:
```suggestion
        INVALID: `Invalid Repository URL. The URL must start with the "http(s)", "git", or "ssh" prefix and end with the ".git" suffix.`,
```

File: core-ui/src/components/Lambdas/constants.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (kazydek) [reply core-ui/src/components/Lambdas/constants.js:174]: 👍 I updated it correctly above.

File: core-ui/src/components/Lambdas/constants.js (hunk 18 above)
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/constants.js]: Updated

**Review by kazydek — APPROVED** [review]

**Review by aerfio — DISMISSED** [review]

**Review by kazydek — DISMISSED** [review]

**Review by Wawrzyn321 — APPROVED** [review]

**Review by aerfio — APPROVED** [review]

**Review by kazydek — APPROVED** [review]

**Comment by magicmatatjahu** [comment]
Merged in https://github.com/kyma-project/console/commit/ed03c5d3c31cc333c715149e22964e8022699b1a

**Pull request closed without merging**